tests/rustdoc-ui/bad-intra-doc.rs

```diff
@@ -0,0 +1,13 @@
+#![no_std]
+#![deny(rustdoc::broken_intra_doc_links)]
+
+// regression test for https://github.com/rust-lang/rust/issues/54191
+
+/// this is not a link to [`example.com`]
```

Comment:
These lines need [error annotations][]. Otherwise, they fail the test even though the stderr file matches.

[error annotations]: https://rustc-dev-guide.rust-lang.org/tests/ui.html#error-annotation-examples

```suggestion
/// this is not a link to [`example.com`] //~ERROR unresolved link
```